### PR TITLE
[4] user log-in endpoint 

### PR DESCRIPTION
**Pull request by Riwuko**
[4] Add ability for users to login

*Description* This branch adds log-in endpoint, using JWT token. There is also detail-view endpoint for logged users which returns user-email.

### CODE REVIEW

File: user_account/api/views.py [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
+class UserLoginView(generics.CreateAPIView):
+    serializer_class = UserAccountSerializer
+    queryset = UserAccount.objects.all()
+    permission_classes = [AllowAny]
+
+    def create(self, request):
+        email = request.data.get("email")
+        password = request.data.get("password")
+
+        user = UserAccount.objects.get(email=email)
```
Comment (nitz14) [inline user_account/api/views.py]: UserAccount.objects.get(email=email) can raise exception if you don't have Account with that email

File: user_account/api/views.py [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
+        if (user is None) or (not user.check_password(password)):
+            raise exceptions.AuthenticationFailed("Wrong email or password")
+
+        payload = jwt_payload_handler(user)
+        token = jwt.encode(payload, settings.SECRET_KEY)
+        user_details = {}
+        user_details["email"] = user.email
+        user_details["token"] = token
```
Comment (nitz14) [inline user_account/api/views.py]:
```suggestion
        user_details = {
           "email": user.email,
           "token": token,
         }
```

File: user_account/api/views.py [hunk 3]
```diff
@@ -1,11 +1,49 @@
...
+        return Response(user_details, status=status.HTTP_200_OK)
```
Comment (nitz14) [inline user_account/api/views.py]: Should this endpoint return email ?

Comment (nitz14) [reply user_account/api/views.py]: please take a look at this :
https://simpleisbetterthancomplex.com/tutorial/2018/12/19/how-to-use-jwt-authentication-with-django-rest-framework.html

Comment (Riwuko) [reply user_account/api/views.py]: Ok, changed! I was thinking we were supposed to write this by ourselfes to practice. Can I just use for example django-allauth for mails and social-login?

File: user_account/api/views.py [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
+    permission_classes = [AllowAny]
+
+
+class UserDetailView(generics.RetrieveUpdateAPIView):
+    serializer_class = UserAccountSerializer
+    queryset = UserAccount.objects.all()
+
+    def get(self, request, *args, **kwargs):
+        serializer = self.serializer_class(request.user)
+        return Response(serializer.data["email"], status=status.HTTP_200_OK)
```
Comment (robertmusil) [inline user_account/api/views.py]: You should pretty much always return data in dict form when working with json apis

File: config/settings.py [hunk 5]
```diff
@@ -48,8 +50,21 @@
...
+    "DEFAULT_PERMISSION_CLASSES": [
+        "rest_framework.permissions.IsAuthenticated",
+    ],
+}
+
+AUTH_USER_MODEL = "user_account.UserAccount"
+
+SIMPLE_JWT = {
+    "ACCESS_TOKEN_LIFETIME": datetime.timedelta(minutes=30),
+    "REFRESH_TOKEN_LIFETIME": relativedelta(years=1),
```
Comment (robertmusil) [inline config/settings.py]: why not simply timedelta = 365 days

File: .env-example [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+ALLOWED_HOSTS=127.0.0.1,localhost
+POSTGRES_DB=abc
+POSTGRES_USER=abc
+POSTGRES_PASSWORD=abc
+POSTGRES_HOST=abc
+POSTGRES_PORT=111
+REFRESH_TOKEN_SECRET=abc
+
+EMAIL_HOST_USER=ddd
+EMAIL_HOST_PASSWORD=dd
```
Comment (nitz14) [inline .env-example]: please add new line

File: user_account/account_manager.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+            raise ValueError("Email can't be empty")
+
+        user = self.model(email=self.normalize_email(email))
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, password):
+        user = self.create_user(email, password)
+        user.is_admin = True
```
Comment (nitz14) [inline user_account/account_manager.py]: its enough ??
what with : 
is_staff, is_superuser

File: user_account/models/__init__.py [hunk 8]
```diff
@@ -0,0 +1 @@
+from user_account.models.user_account import UserAccount
```
Comment (nitz14) [inline user_account/models/__init__.py]: missing empty line in the end of file

File: user_account/migrations/0002_auto_20200923_1021.py [hunk 9]
```diff
@@ -1,18 +0,0 @@
-# Generated by Django 2.2.16 on 2020-09-23 08:21
```
Comment (nitz14) [inline user_account/migrations/0002_auto_20200923_1021.py:1]: we shouldn't delete migrations.

File: user_account/account_manager.py [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+        user.save()
+        return user
+
+    def create_superuser(self, email, password, **extra_fields):
+        extra_fields.setdefault("is_staff", True)
+        extra_fields.setdefault("is_superuser", True)
+        extra_fields.setdefault("is_active", True)
+        user = self.create_user(email, password, **extra_fields)
+        user.save()
+        return user
```
Comment (nitz14) [inline user_account/account_manager.py]:
```suggestion
        return self.create_user(email, password, **extra_fields)
```

you don't need to save this object again

**Review by nitz14 — APPROVED** [review]

**Review by robertmusil — APPROVED** [review]

**Pull request merged**